app/components/map-component/index.tsx

```diff
@@ -0,0 +1,210 @@
+import { makeStyles, useTheme } from "@rneui/themed"
+import MapView, { Region, MapMarker as MapMarkerType } from "react-native-maps"
+import MapStyles from "./map-styles.json"
+import React, { useRef } from "react"
+import { BusinessMapMarkersQuery, MapMarker } from "@app/graphql/generated"
+import { useI18nContext } from "@app/i18n/i18n-react"
+import MapMarkerComponent from "../map-marker-component"
+import { PermissionStatus, RESULTS, request } from "react-native-permissions"
+import { LOCATION_PERMISSION, getUserRegion } from "@app/screens/map-screen/map-screen"
+import LocationButtonCopy from "./location-button-copy"
+import debounce from "lodash.debounce"
+import { updateMapLastCoords } from "@app/graphql/client-only-query"
+import { useApolloClient } from "@apollo/client"
+import { OpenSettingsElement, OpenSettingsModal } from "./open-settings-modal"
+import { View } from "react-native"
+import { isIOS } from "@rneui/base"
+
+type Props = {
+  data?: BusinessMapMarkersQuery
+  userLocation?: Region
+  permissionsStatus?: PermissionStatus
+  setPermissionsStatus: (_: PermissionStatus) => void
+  handleMapPress: () => void
+  handleMarkerPress: (_: MapMarker) => void
+  focusedMarker: MapMarker | null
+  focusedMarkerRef: React.MutableRefObject<MapMarkerType | null>
+  handleCalloutPress: (_: MapMarker) => void
+  onError: () => void
+}
+
+export default function MapComponent({
+  data,
+  userLocation,
+  permissionsStatus,
+  setPermissionsStatus,
+  handleMapPress,
+  handleMarkerPress,
+  focusedMarker,
+  focusedMarkerRef,
+  handleCalloutPress,
+  onError,
```

Comment:
does `onError` needs to be pass as a prop? is it only ever specific to `alertOnLocationError`?